gapis/trace/android/trace.go

```diff
@@ -150,45 +151,52 @@ func (t *androidTracer) ProcessProfilingData(ctx context.Context, buffer *bytes.
 	return nil, log.Errf(ctx, nil, "Failed to process Perfetto trace for device %v", gpuName)
 }
 
-func (t *androidTracer) Validate(ctx context.Context) error {
+func (t *androidTracer) Validate(ctx context.Context, enableLocalFiles bool) (*service.DeviceValidationResult, error) {
 	ctx = status.Start(ctx, "Android Device Validation")
 	defer status.Finish(ctx)
+
 	if t.v == nil {
-		return log.Errf(ctx, nil, "No validator found for device %d", t.b.Instance().ID.ID())
+		return nil, log.Errf(ctx, nil, "No validator found for device %d", t.b.Instance().ID.ID())
 	}
 	d := t.b.(adb.Device)
 	osConfiguration := d.Instance().GetConfiguration()
 	if osConfiguration.GetOS().GetAPIVersion() < minimumSupportedApiLevel {
-		return log.Errf(ctx, nil, "Unsupported OS version on device %d", d.Instance().ID.ID())
+		return &service.DeviceValidationResult{
+			ValidationErrorMsg: fmt.Sprintf("Unsupported OS version on device %d", d.Instance().ID.ID()),
+		}, nil
 	}
 	if osConfiguration.GetPerfettoCapability() == nil {
-		return log.Errf(ctx, nil, "No Perfetto Capability found on device %d", d.Instance().ID.ID())
+		return &service.DeviceValidationResult{
+			ValidationErrorMsg: fmt.Sprintf("No Perfetto Capability found on device %d", d.Instance().ID.ID()),
+		}, nil
 	}
 	if gpuProfiling := osConfiguration.GetPerfettoCapability().GetGpuProfiling(); gpuProfiling == nil || gpuProfiling.GetGpuCounterDescriptor() == nil {
-		return log.Errf(ctx, nil, "No GPU profiling capabilities found on device %d", d.Instance().ID.ID())
+		return &service.DeviceValidationResult{
+			ValidationErrorMsg: fmt.Sprintf("No GPU profiling capabilities found on device %d", d.Instance().ID.ID()),
```

Comment:
"The device does not support GPU profiling or profiling support was not detected".